modules/kafka/src/main/scala/com/snowplowanalytics/snowplow/streams/kafka/source/KafkaSource.scala

```diff
@@ -35,57 +35,57 @@ private[kafka] object KafkaSource {
   def build[F[_]: Async](
     config: KafkaSourceConfig,
     authHandlerClass: String
-  ): F[SourceAndAck[F]] =
-    LowLevelSource.toSourceAndAck(lowLevel(config, authHandlerClass))
+  ): Resource[F, SourceAndAck[F]] =
+    for {
+      kafkaConsumer <- KafkaConsumer.resource(consumerSettings[F](config, authHandlerClass))
+      result <- Resource.eval(LowLevelSource.toSourceAndAck(lowLevel(config, kafkaConsumer)))
+    } yield result
 
   private def lowLevel[F[_]: Async](
     config: KafkaSourceConfig,
-    authHandlerClass: String
-  ): LowLevelSource[F, KafkaCheckpoints[F]] =
-    new LowLevelSource[F, KafkaCheckpoints[F]] {
-      def checkpointer: Checkpointer[F, KafkaCheckpoints[F]] = kafkaCheckpointer
+    kafkaConsumer: KafkaConsumer[F, Array[Byte], ByteBuffer]
+  ): LowLevelSource[F, KafkaCheckpoints] =
+    new LowLevelSource[F, KafkaCheckpoints] {
+      def checkpointer: Checkpointer[F, KafkaCheckpoints] = kafkaCheckpointer(kafkaConsumer)
 
-      def stream: Stream[F, Stream[F, Option[LowLevelEvents[KafkaCheckpoints[F]]]]] =
-        kafkaStream(config, authHandlerClass)
+      def stream: Stream[F, Stream[F, Option[LowLevelEvents[KafkaCheckpoints]]]] =
+        kafkaStream(config, kafkaConsumer)
 
       def debounceCheckpoints: FiniteDuration = config.debounceCommitOffsets
     }
 
-  case class OffsetAndCommit[F[_]](offset: Long, commit: F[Unit])
-  case class KafkaCheckpoints[F[_]](byPartition: Map[Int, OffsetAndCommit[F]])
+  type KafkaCheckpoints = Map[TopicPartition, OffsetAndMetadata]
 
-  private implicit def offsetAndCommitSemigroup[F[_]]: Semigroup[OffsetAndCommit[F]] = new Semigroup[OffsetAndCommit[F]] {
-    def combine(x: OffsetAndCommit[F], y: OffsetAndCommit[F]): OffsetAndCommit[F] =
+  private implicit def offsetAndMetadataSemigroup: Semigroup[OffsetAndMetadata] = new Semigroup[OffsetAndMetadata] {
+    def combine(x: OffsetAndMetadata, y: OffsetAndMetadata): OffsetAndMetadata =
       if (x.offset > y.offset) x else y
   }
 
-  private def kafkaCheckpointer[F[_]: Async]: Checkpointer[F, KafkaCheckpoints[F]] = new Checkpointer[F, KafkaCheckpoints[F]] {
-    def combine(x: KafkaCheckpoints[F], y: KafkaCheckpoints[F]): KafkaCheckpoints[F] =
-      KafkaCheckpoints(x.byPartition |+| y.byPartition)
+  private def kafkaCheckpointer[F[_]: Async](kafkaConsumer: KafkaConsumer[F, Array[Byte], ByteBuffer]): Checkpointer[F, KafkaCheckpoints] =
+    new Checkpointer[F, KafkaCheckpoints] {
+      def combine(x: KafkaCheckpoints, y: KafkaCheckpoints): KafkaCheckpoints =
+        x |+| y
 
-    val empty: KafkaCheckpoints[F] = KafkaCheckpoints(Map.empty)
-    def ack(c: KafkaCheckpoints[F]): F[Unit]  = c.byPartition.values.toList.parTraverse(_.commit).void
-    def nack(c: KafkaCheckpoints[F]): F[Unit] = Applicative[F].unit
-  }
+      val empty: KafkaCheckpoints = Map.empty
+      def ack(c: KafkaCheckpoints): F[Unit] =
+        kafkaConsumer.commitSync(c)
```

Comment:
This is the most important line of this PR.  Everything else is just setting us up so we can call `kafkaConsumer.commitSync(c)`.